src/node/request.ts

```diff
@@ -0,0 +1,74 @@
+const https = require('https');
+const http = require('http');
+
+import type { ClientRequest, RequestOptions, ServerResponse } from 'http';
+import { isDev } from '../env';
+
+const protocol = isDev ? http : https;
+
+const options = {
+  hostname: import.meta.env.VITE_API_BASE_URL,
+  port: isDev ? 8080 : undefined,
+  headers: { 'Content-Type': 'application/json' },
+};
+
+function request(options: RequestOptions) {
+  return new Promise((resolve, reject) => {
+    const req: ClientRequest = protocol.request(
+      options,
+      (res: ServerResponse) => {
+        let data = '';
+
+        // Collect the response data
+        res.on('data', (chunk) => {
+          data += chunk;
+        });
+
+        // Resolve or Reject the Promise when the response ends
+        res.on('end', () => {
+          const { statusCode } = res;
+          if (statusCode >= 200 && statusCode < 300) {
```

Comment:
* why less than 300? 3xx should be fine or?
* are requests auto forwarded?
* imo `statusCode < 400` should be fine here